Sources/MongoSwift/MongoClient.swift

```diff
@@ -253,6 +267,23 @@ public class MongoClient {
         )
     }
 
+    deinit {
+        assert(self.isClosed, "MongoClient was not closed before deinitialization")
+    }
+
+    /// Closes this `MongoClient`. Call this method exactly once when you are finished using the client. You must
+    /// ensure that all operations using the client have completed before calling this. The returned future must be
+    /// fulfilled before the client goes out of scope.
+    public func close() -> EventLoopFuture<Void> {
+        return self.operationExecutor.execute {
+            self.connectionPool.close()
+            self.isClosed = true
```

Comment:
this is technically kind of racey since you could call this from one thread while still using the client in another, but I think the requirements I've documented (call it exactly once, when everything else is done) make it acceptable.

I'm not sure how often people will even *really* need to call this method since I imagine people will often just have one client that for the whole application and only clean it up when their app is shutting down...